### DISCUSSION TITLE
A JS-Engine Button

### DISCUSSION THREAD

**Discussion by giittyy** [mProjectsCode/obsidian-meta-bind-plugin#218] [Q&A]
Has anyone create a button using JS-Engine. I can't find an example anywhere, however it is in the api. I don't want to run JS from a file.

**Comment by Jacks-01** [comment]
Can you give more detail? If you use a [JS View Field](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/viewfields/#creating-js-view-fields), you can use JS-Engine syntax to create a button in the bottom half of the codeblock (the js part).

```meta-bind-js-view
{HP_CURRENT} as HP_CURRENT
{HP_TOTAL} as HP_TOTAL
---
let markdownBuilder = engine.markdown.createBuilder();

let long_rest_button = markdownBuilder.createCodeBlock('meta-bind-button',
`label: Long Rest
hidden: true
class: ""
tooltip: ""
id: "long-rest"
style: primary
actions:
  - type: updateMetadata
    bindTarget: HP_CURRENT
    evaluate: false
    value: "${context.bound.HP_TOTAL}"`)

return markdownBuilder
```

**Reply by mProjectsCode** [reply]
Or you use a JS engine code block, if you don't need anything dependent on metadata.
You can also use the API methods to create the button, instead of the markdown rendering feature.

**Reply by Jacks-01** [reply]
What API method are you referring to? I didn't see anything in the documentation for creating buttons. Or a super basic example if thats not too much.

**Reply by mProjectsCode** [reply]
https://github.com/mProjectsCode/obsidian-meta-bind-plugin/blob/eff97ba18f3af60a32d17e8a3686b309b0b7cd14/src/api/ObsidianAPI.ts#L44

https://github.com/mProjectsCode/obsidian-meta-bind-plugin/blob/eff97ba18f3af60a32d17e8a3686b309b0b7cd14/src/api/ObsidianAPI.ts#L220

These two API methods should work.
And yes the API is not documented, as I am to lazy to do that and you should be able to read the source code if you want to use them.
